cli/CommandExecutor.cpp

```diff
@@ -0,0 +1,200 @@
+/**
+ *   Copyright 2011-2015 Quickstep Technologies LLC.
+ *   Copyright 2015-2016 Pivotal Software, Inc.
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "cli/CommandExecutor.hpp"
+
+#include <cstddef>
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "catalog/CatalogAttribute.hpp"
+#include "catalog/CatalogDatabase.hpp"
+#include "catalog/CatalogRelation.hpp"
+#include "catalog/CatalogRelationSchema.hpp"
+#include "cli/PrintToScreen.hpp"
+#include "parser/ParseStatement.hpp"
+#include "utility/PtrVector.hpp"
+#include "utility/Macros.hpp"
+#include "utility/SqlError.hpp"
+
+#include "gflags/gflags.h"
+#include "glog/logging.h"
+
+using std::fprintf;
+using std::fputc;
+using std::fputs;
+using std::size_t;
+using std::vector;
+using std::string;
+
+namespace quickstep {
+
+void CommandExecutor::executeCommand(const ParseStatement &statement,
+                                     const CatalogDatabase *catalog_database,
+                                     FILE *out) {
+  const ParseCommand &command = static_cast<const ParseCommand &>(statement);
+  const PtrVector<ParseString> *arguments = command.arguments();
+  std::string command_str = command.command()->value();
+  if (command_str.compare("\\dt") == 0) {
+    executeDescribeDatabase(arguments, catalog_database, out);
+  } else if (command_str.compare("\\d") == 0) {
```

Comment:
@zuyu 

Sure. I did think about giving names to the command. The ParseCommand class  returns  the command string. The only way to know what the command is would be  from the command string.  
Do you have any suggestions on what would be a better design of mapping the command strings to the names and the resolving the names from the command string that comes from the parser?